File: wagerproof-ios-native/Wagerproof/Features/LearnMore/Components/Slide1_Create247Agent.swift
import SwiftUI
import WagerproofDesign

/// Mirrors `wagerproof-mobile/components/learn-wagerproof/slides/Slide1_Create247Agent.tsx`.
///
/// The RN version optionally renders a Lottie `RobotAnalyzing.json`. We don't
/// bundle Lottie in the Swift port (no dependency yet) — instead we render a
/// large SF Symbol robot inside the hero card with the same gentle pulse the
/// Lottie scene used. See `// FIDELITY-WAIVER #063` below.
struct Slide1_Create247Agent: View {
    @State private var pulse: Bool = false

    private struct Bullet: Identifiable {
        let id = UUID()
        let systemImage: String
        let title: String
        let description: String
    }

    private let bullets: [Bullet] = [
        Bullet(
            systemImage: "brain.head.profile",
            title: "Build multiple agents",
            description: "Create as many agents as you want, each with a different betting strategy."
        ),
        Bullet(
            systemImage: "clock.fill",
            title: "24/7 research",
            description: "Your agents continuously research games and surface picks around the clock."
        ),
        Bullet(
            systemImage: "trophy.fill",
            title: "Global leaderboard",
            description: "View the world's best agents, their records, and their latest picks."
        ),
    ]

    var body: some View {
        VStack(spacing: Spacing.sm) {
            // Hero animation card — Lottie placeholder.
            // FIDELITY-WAIVER #063 — RN renders `RobotAnalyzing.json`; the Swift
            // port draws the static SF Symbol robot with a gentle scale pulse.
            ZStack {
                Image(systemName: "brain.head.profile")
                    .resizable()
                    .scaledToFit()
                    .foregroundStyle(Color.appPrimary)
                    .frame(width: 72, height: 72)
                    .scaleEffect(pulse ? 1.05 : 0.95)
                    .animation(.easeInOut(duration: 1.6).repeatForever(autoreverses: true), value: pulse)
            }
            .frame(maxWidth: .infinity, minHeight: 150)
            .background(Color.appSurfaceElevated)
            .clipShape(RoundedRectangle(cornerRadius: 14, style: .continuous))
            .onAppear { pulse = true }

            VStack(spacing: Spacing.sm) {
                ForEach(bullets) { bullet in
                    bulletCard(bullet)
                }
            }
        }
    }

    private func bulletCard(_ bullet: Bullet) -> some View {
        VStack(alignment: .leading, spacing: 4) {
            HStack(spacing: Spacing.sm) {
                ZStack {
                    Circle().fill(Color.appPrimary.opacity(0.14))
                    Image(systemName: bullet.systemImage)
                        .font(.system(size: 14, weight: .semibold))
                        .foregroundStyle(Color.appPrimary)
                }
                .frame(width: 26, height: 26)

                Text(bullet.title)
                    .font(.system(size: 13, weight: .bold))
                    .foregroundStyle(Color.appTextPrimary)
            }
            Text(bullet.description)
                .font(.system(size: 11))
                .lineSpacing(2)
                .foregroundStyle(Color.appTextSecondary)
                .fixedSize(horizontal: false, vertical: true)
        }
        .padding(10)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(Color.appSurfaceElevated)
        .clipShape(RoundedRectangle(cornerRadius: 12, style: .continuous))
    }
}

File: wagerproof-android-native/app/src/main/java/com/wagerproof/app/features/learn/slides/Slide1Create247Agent.kt
package com.wagerproof.app.features.learn.slides

import androidx.compose.animation.core.RepeatMode
import androidx.compose.animation.core.animateFloat
import androidx.compose.animation.core.infiniteRepeatable
import androidx.compose.animation.core.rememberInfiniteTransition
import androidx.compose.animation.core.tween
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.heightIn
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.scale
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.wagerproof.core.design.icons.AppIcon
import com.wagerproof.core.design.tokens.AppColors
import com.wagerproof.core.design.tokens.Spacing

/**
 * Port of iOS `Slide1_Create247Agent.swift`.
 *
 * FIDELITY-WAIVER #063 — iOS/RN render a Lottie robot; the port draws a static
 * SF-symbol robot with a gentle scale pulse (established Lottie-replacement waiver).
 */
@Composable
fun Slide1Create247Agent(modifier: Modifier = Modifier) {
    val transition = rememberInfiniteTransition(label = "robotPulse")
    val pulse by transition.animateFloat(
        initialValue = 0.95f,
        targetValue = 1.05f,
        animationSpec = infiniteRepeatable(tween(1600), RepeatMode.Reverse),
        label = "pulse",
    )

    Column(modifier, verticalArrangement = Arrangement.spacedBy(Spacing.sm)) {
        Box(
            Modifier
                .fillMaxWidth()
                .heightIn(min = 150.dp)
                .clip(RoundedCornerShape(14.dp))
                .background(AppColors.appSurfaceElevated),
            contentAlignment = Alignment.Center,
        ) {
            Icon(
                imageVector = AppIcon.fromSystemName("brain.head.profile")!!.imageVector,
                contentDescription = null,
                tint = AppColors.appPrimary,
                modifier = Modifier.size(72.dp).scale(pulse),
            )
        }

        Column(verticalArrangement = Arrangement.spacedBy(Spacing.sm)) {
            bullets.forEach { bulletCard(it) }
        }
    }
}

private data class Bullet(val icon: ImageVector, val title: String, val description: String)

private val bullets: List<Bullet> = listOf(
    Bullet(
        AppIcon.fromSystemName("brain.head.profile")!!.imageVector,
        "Build multiple agents",
        "Create as many agents as you want, each with a different betting strategy.",
    ),
    Bullet(
        AppIcon.fromSystemName("clock.fill")!!.imageVector,
        "24/7 research",
        "Your agents continuously research games and surface picks around the clock.",
    ),
    Bullet(
        AppIcon.fromSystemName("trophy.fill")!!.imageVector,
        "Global leaderboard",
        "View the world's best agents, their records, and their latest picks.",
    ),
)

@Composable
private fun bulletCard(bullet: Bullet) {
    Column(
        Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(12.dp))
            .background(AppColors.appSurfaceElevated)
            .padding(10.dp),
        verticalArrangement = Arrangement.spacedBy(4.dp),
    ) {
        Row(
            horizontalArrangement = Arrangement.spacedBy(Spacing.sm),
            verticalAlignment = Alignment.CenterVertically,
        ) {
            Box(
                Modifier
                    .size(26.dp)
                    .clip(CircleShape)
                    .background(AppColors.appPrimary.copy(alpha = 0.14f)),
                contentAlignment = Alignment.Center,
            ) {
                Icon(
                    imageVector = bullet.icon,
                    contentDescription = null,
                    tint = AppColors.appPrimary,
                    modifier = Modifier.size(14.dp),
                )
            }
            Text(bullet.title, fontSize = 13.sp, fontWeight = FontWeight.Bold, color = AppColors.appTextPrimary)
        }
        Text(bullet.description, fontSize = 11.sp, color = AppColors.appTextSecondary, lineHeight = 15.sp)
    }
}
